src/wallet/coinselection.cpp

```diff
@@ -84,11 +84,11 @@ bool SelectCoinsBnB(std::vector<OutputGroup>& utxo_pool, const CAmount& selectio
     std::sort(utxo_pool.begin(), utxo_pool.end(), descending);
 
     CAmount curr_waste = 0;
-    std::vector<bool> best_selection;
+    std::vector<size_t> best_selection;
     CAmount best_waste = MAX_MONEY;
 
     // Depth First search loop for choosing the UTXOs
-    for (size_t i = 0; i < TOTAL_TRIES; ++i) {
+    for (size_t i = 0, curr_index = 0; i < TOTAL_TRIES; ++i, ++curr_index) {
```

Comment:
Optional: given that we now have two indices incremented in this for-loop, perhaps rename `i` to `try` and `curr_index` to `utxo_pool_depth`.

Reply:
Done, though I went with `utxo_pool_index`